Instruments/power_control.py

```diff
@@ -61,21 +61,19 @@ def get(self, n_slow_tries = 0, slow_wait = 0.5):
         if not success: raise ValueError("no response after 30 tries!!")
         return data
     def get_bytes(self,ending):
-        data = self.sock.recv(1024)
-        success = False
-        for j in range(300):
-            print("I've got %d bytes"%len(data))
-            if len(data) == 0:
+        data = ""
+        while not data.endswith(ending):
+            counter = 0
+            new_data = ""
+            while not len(new_data) > 0:
+                new_data = self.sock.recv(1024)
+                print("new data is %d bytes"%len(new_data))
+                counter += 1
+                if counter > 300:
+                    raise ValueError("No data acquired after 300 tries!!")
                 time.sleep(0.01)
-                data += self.sock.recv(1024)
-            else:
-                if data.endswith(ending):
-                    success = True
-                    break
-                else:
-                    data += self.sock.recv(1024)
-        if not success: raise ValueError("no success after 300 tries!!")
-        return data
+                data += new_data
```

Comment:
at this point, you would break out of the while look on line 65, unless you reset `new_data=""`